crates/uv-requirements/src/specification.rs

```diff
@@ -221,6 +161,62 @@ impl RequirementsSpecification {
         })
     }
 
+    /// Attempt to read metadata from the `pyproject.toml` directly.
+    ///
+    /// Since we only use this path for directly included pyproject.toml, we are strict about
+    /// PEP 621.
```

Comment:
What does it mean to be "strict about PEP 621" here? Can you expand on this comment? E.g., we're still falling back to treating it as a source tree below.